src/model/modifier/MediaUtils.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule MediaUtils
+ * @typechecks
+ * @flow
+ */
+
+'use strict';
+
+const BlockMapBuilder = require('BlockMapBuilder');
+const CharacterMetadata = require('CharacterMetadata');
+const ContentBlock = require('ContentBlock');
+const DraftModifier = require('DraftModifier');
+const EditorState = require('EditorState');
+const {List, Repeat} = require('immutable');
+
+const genKey = require('generateBlockKey');
+
+const MediaUtils = {
+  insertMedia: function(
+    editorState: EditorState,
+    entityKey: string,
+    character: string
+  ): EditorState {
+    const contentState = editorState.getCurrentContent();
+    const selectionState = editorState.getSelection();
+
+    const afterRemoval = DraftModifier.removeRange(
+      contentState,
+      selectionState,
+      'backward'
+    );
+
+    const targetSelection = afterRemoval.getSelectionAfter();
+    const afterSplit = DraftModifier.splitBlock(afterRemoval, targetSelection);
```

Comment:
Is the need to split the block here in case the current block has text in it still? In the case that it's completely empty, might be better to not split the block, so that it doesn't end up as an extra empty block above the media block. Just a thought!